app/Controllers/Http/ProfilesController.ts

```diff
@@ -0,0 +1,87 @@
+// import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Profile from 'App/Models/Profile';
+import { schema, rules } from '@ioc:Adonis/Core/Validator';
+export default class ProfilesController {
+    public async addProfile({ auth, request, response }) {
+        await auth.use('api').check()
+        if (auth.use('api').isLoggedIn) {
+            const newUserSchema = schema.create({
+                name: schema.string([
+                    rules.minLength(3),
+                    rules.maxLength(30),
+                ]),
+                email: schema.string([
+                    rules.email()
+                ]),
+                gender: schema.string([]),
+                dob: schema.date.optional({
+                    format: 'dd-MM-yyyy',
+                }),
+                user_id: schema.number([]),
```

Comment:
Instead of passing `user_id` in request, we can use `auth.user` directly to get the user id.